python/cuml/test/test_linear_model.py

```diff
@@ -128,6 +131,41 @@ def test_linear_regression_model(datatype, algorithm, nrows, column_info):
         assert array_equal(skols_predict, cuols_predict, 1e-1, with_sign=True)
 
 
+@pytest.mark.parametrize("datatype", [np.float32, np.float64])
+@pytest.mark.parametrize("algorithm", ["eig", "svd", "qr", "svd-qr"])
+@pytest.mark.parametrize("fit_intercept", [True, False])
+@pytest.mark.parametrize("normalize", [True, False])
+def test_weighted_linear_regression(datatype, algorithm, fit_intercept,
+                                    normalize):
+    nrows, ncols, n_info = 1000, 20, 10
+    max_weight = 10
+    noise = 20
+    X_train, X_test, y_train, y_test = make_regression_dataset(
+        datatype, nrows, ncols, n_info, noise=noise
+    )
+
+    # set weight per sample to be from 1 to max_weight
+    wt = np.random.randint(1, high=max_weight, size=len(X_train))
```

Comment:
It might not be a bad idea to try a couple different distributions of sample weights, such as a couple highly skewed distributions.